store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -367,9 +384,8 @@ private CarbonTable buildCarbonTable() {
       //  user passed size 4 but supplied only 2 fileds
       for (Field field : schema.getFields()) {
         if (null != field) {
-          if (field.getDataType() == DataTypes.STRING ||
-              field.getDataType() == DataTypes.DATE ||
-              field.getDataType() == DataTypes.TIMESTAMP) {
+          if (field.getDataType() == DataTypes.STRING || field.getDataType() == DataTypes.DATE
```

Comment:
Sort column is also supported for long, short, int data type , add a reverse check if data type is not of  double, bigdecimal, struct, array then add to sort column

Reply:
This check is for creating all the dimension columns as sort columns when by default sort column is not specified.